srv/review-service.cds

```diff
@@ -0,0 +1,50 @@
+using {my.bookshop as my} from '../db/index';
+
+@path : 'review'
+service ReviewService {
+    entity Reviews as projection on my.Reviews;
+
+    @readonly
+    entity Books   as projection on my.Books excluding {
+        createdBy,
+        modifiedBy
+    }
+
+    @readonly
+    entity Authors as projection on my.Authors;
+
+    // input validation
+    annotate Reviews with {
+        subject @mandatory;
+        title   @mandatory;
+        rating  @assert.enum;
+    }
+
+    // access control restrictions
+    annotate ReviewService.Reviews with @restrict : [
+    {
```

Comment:
Why not change it that the authenticated-user can only read its own reviews and the admin can read all?
We could do it something like that: `grant * to authenticated-user where reviewer = $user OR grant * to admin`

Reply:
Yeah, that makes sense for the ReviewService. I'll change that.